components/host-dashboard/CollectiveSettingsModal.js

```diff
@@ -67,7 +69,10 @@ const CollectiveSettingsModal = ({ host, collective, ...props }) => {
   const [selectedOption, setSelectedOption] = useState(
     hostFeePercent === host.hostFeePercent ? HOST_FEE_STRUCTURE.DEFAULT : HOST_FEE_STRUCTURE.CUSTOM_FEE,
   );
-  const [hasGrant, setHasGrant] = useState(collective?.settings?.expenseTypes?.hasGrant || false);
+  const isProjectOrFund = [CollectiveType.FUND, CollectiveType.PROJECT].includes(collective.type);
+  const [hasGrant, setHasGrant] = useState(
+    accountSupportsGrants(collective?.settings, host?.settings) || isProjectOrFund,
+  );
```

Comment:
Not blocking on this because we really need to merge a fix but it's sad that we did not implement the helper with the API suggested in https://github.com/opencollective/opencollective-frontend/pull/7090#discussion_r745337346, that would have reduced complexity and code duplication.

```suggestion
  const [hasGrant, setHasGrant] = useState(accountSupportsGrants(collective));
```

Reply:
Oh, I didn't think we need to implement this on the API side. Let me reiterate this on a separate PR. 🤔 🙂

Reply:
Nono, it's not to be implemented on the API side, it's for the fronted. What I meant by `t's sad that we did not implement the helper with the API suggested in #7090 (comment)` is literally `application programming interface`, in other words I was hoping we could do something like this for the prototype of the function:

```es6
accountSupportsGrants = (account): boolean => {}
```

Reply:
@Betree : Ops sorry, I was confused with this. But I've opened a new PR; https://github.com/opencollective/opencollective-frontend/pull/7124 which simplifies matters. Now I've included an optional argument for the function to take the host as well, since there's one place, 

https://github.com/opencollective/opencollective-frontend/blob/9a05d7b2afea21b7b00f10983236340c1c0eee23/components/host-dashboard/CollectiveSettingsModal.js#L66-L73

where the `collective` and `host` are separately passed to the component and the `collective` object doesn't include the `host` fields.